### PR TITLE
fix: TCP connections leaking after RTC disconnects

### PR DESCRIPTION
**Pull request by kylecarbs**

### CODE REVIEW

**Comment by kylecarbs** [comment]
[ch15285]

File: wsnet/dial_test.go [hunk 1]
```diff
@@ -266,6 +266,46 @@ func TestDial(t *testing.T) {
...
+
+		closeTurn()
+
+		for i := 0; i < 15; i++ {
+			time.Sleep(time.Second)
+
+			if len(l.(*listener).connClosers) == 0 {
+				break
+			}
+		}
```
Comment (deansheather) [inline wsnet/dial_test.go:305]: Use `assert.Eventually`

File: wsnet/listen.go [hunk 2]
```diff
@@ -255,17 +255,26 @@ func (l *listener) negotiate(ctx context.Context, conn net.Conn) {
...
+				l.log.Info(ctx, "connection state change", slog.F("state", pcs.String()))
 				if pcs == webrtc.PeerConnectionStateConnecting {
 					return
 				}
-				_ = conn.Close()
+				if pcs == webrtc.PeerConnectionStateConnected {
+					// Safe to close the negotiating WebSocket.
+					_ = conn.Close()
+					return
+				}
```
Comment (deansheather) [inline wsnet/listen.go:266]: nit: switch statement

File: wsnet/listen.go [hunk 3]
```diff
@@ -255,17 +255,26 @@ func (l *listener) negotiate(ctx context.Context, conn net.Conn) {
...
 			rtc.OnConnectionStateChange(func(pcs webrtc.PeerConnectionState) {
-				l.log.Debug(ctx, "connection state change", slog.F("state", pcs.String()))
+				l.log.Info(ctx, "connection state change", slog.F("state", pcs.String()))
 				if pcs == webrtc.PeerConnectionStateConnecting {
 					return
 				}
-				_ = conn.Close()
+				if pcs == webrtc.PeerConnectionStateConnected {
+					// Safe to close the negotiating WebSocket.
+					_ = conn.Close()
```
Comment (deansheather) [inline wsnet/listen.go:264]: Nice

**Review by deansheather — APPROVED** [review]

**Pull request merged**